EIPS/eip-223.md

```diff
@@ -0,0 +1,368 @@
+---
+eip: 223
+title: 223 Token with communication model
+description: Token with event handling and communication model
+author: Dexaran (@Dexaran)
+discussions-to: https://ethereum-magicians.org/t/erc-223-token-standard/12894
+status: Review
+type: Standards Track
+category: ERC
+created: 2017-05-03
+---
+
+## Abstract
+
+The following describes an interface for fungible tokens that supports a `tokenReceived` callback to notify contract recipients when tokens are received.
+
+## Motivation
+
+This token introduces a communication model for contracts that can be utilized to straighten the behavior of contracts that interact with such tokens. Specifically, this proposal:
+
+1. Informs receiving contracts of tokens, as opposed to [ERC-20](./eip-20.md) where the recipient of a token transfer gets no notification.
+2. Is more gas-efficient when depositing tokens to contracts.
+3. Allows for `_data` recording for financial transfers.
+
+
+## Specification
+
+
+### Methods
+
+Contracts intending to receive these tokens MUST implement `tokenReceived`.
+
+Tokens transferred to contracts not implementing `tokenReceived` as described below MUST revert.
+
+
+#### `totalSupply`
+
+```solidity
+function totalSupply() view returns (uint256 totalSupply)
+```
+
+Get the total token supply
+
+#### `name`
+
+```solidity
+function name() view returns (string _name)
```

Comment:
Since we're creating a totally new standard, might it make sense to also require a `uri()` function to get EIP-721-like data? This way, the `name()` and `symbol()` functions can also be removed.

Reply:
These functions do not appear to be indicated as optional.

Reply:
fixed